staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode_test.go

```diff
@@ -288,16 +575,32 @@ func TestDecode(t *testing.T) {
 			}
 
 			t.Run(fmt.Sprintf("mode=%s/%s", modeName, tc.name), func(t *testing.T) {
+				if tc.failureReason != "" {
+					t.Skip(tc.failureReason)
+				}
+
 				var dst reflect.Value
 				if tc.into == nil {
 					var i interface{}
 					dst = reflect.ValueOf(&i)
 				} else {
 					dst = reflect.New(reflect.TypeOf(tc.into))
 				}
+
 				err := decMode.Unmarshal(tc.in, dst.Interface())
-				tc.assertOnError(t, err)
-				if tc.want != nil {
+
+				if tc.unexportedError != "" {
+					if err == nil {
+						t.Error("Expected non-nil error: " + tc.unexportedError)
+					} else if !strings.Contains(err.Error(), tc.unexportedError) {
+						t.Errorf("Expected error: %s, got: %s \n", tc.unexportedError, err.Error())
+					}
+					return
+				} else {
```

Comment:
Instead of a new field and a special case here, this can be implemented as an anonymous `func(t *testing.T, e error)` in the existing `assertOnError` field.

Reply:
Fixed.